tests/test_pseudo_points.cc

```diff
@@ -0,0 +1,206 @@
+                                   Fix) {
+    const bool verbose{false};
+    auto & sparse_points = Fix::sparse_points;
+
+    typename Fix::ManagerCollection_t managers{Fix::adaptors};
+    managers.add_structures(Fix::filename, 0, 3);
+    typename Fix::Representation_t representation{Fix::hypers};
+
+    representation.compute(managers);
+
+    std::vector<std::vector<int>> selected_ids;
+    std::random_device rd;
+    std::mt19937 gen{rd()};
+    auto dice{std::uniform_real_distribution<double>(0., 1.)};
+    for (auto & manager : managers) {
+      selected_ids.emplace_back();
+      int ii{0};
+      for (auto center : manager) {
+        (void)center;  // to avoid compiler warning
+        double roll{dice(gen)};
+        if (roll < 0.85) {
+          selected_ids.back().push_back(ii);
+        } else if (verbose) {
+          std::cout << "Center " << ii << " will not be considered."
+                    << std::endl;
+        }
+        ++ii;
+      }
+    }
+
+    sparse_points.push_back(representation, managers, selected_ids);
+
+    auto feat_ref = managers.get_features(representation);
+
+    auto feat_test = sparse_points.get_features();
+
+    for (int i_row{0}; i_row < feat_test.rows(); i_row++) {
+      auto diffs = (feat_ref.rowwise() - feat_test.row(i_row))
+                       .rowwise()
+                       .template lpNorm<1>();
+      // check that one row of feat_ref matches the current row of feat_test
+      BOOST_TEST((diffs.array() < 1e-16).count() == 1);
+
+      if (verbose and (diffs.array() < 1e-16).count() != 1) {  // NOLINT
+        std::cout << "Number of matching row " << i_row << " :"
+                  << (diffs.array() < 1e-16).count() << std::endl;
+        std::cout << feat_ref << std::endl;
+        std::cout << "============================" << std::endl;
+        std::cout << feat_test.row(i_row) << std::endl;
+        std::cout << "####################################" << std::endl;
+      }
+    }
+  }
+
+  /**
+   * Tests the serialization works properly and that trying to get a type
+   * mismatched with the serialized pseudo_points throws an error.
+   */
+  BOOST_FIXTURE_TEST_CASE_TEMPLATE(serialization_test, Fix, multiple_fixtures,
+                                   Fix) {
+    const bool verbose{false};
+    auto & sparse_points = Fix::sparse_points;
+
+    typename Fix::ManagerCollection_t managers{Fix::adaptors};
+    managers.add_structures(Fix::filename, 0, 3);
+    typename Fix::Representation_t representation{Fix::hypers};
+
+    representation.compute(managers);
+
+    std::vector<std::vector<int>> selected_ids;
+    std::random_device rd;
+    std::mt19937 gen{rd()};
+    auto dice{std::uniform_real_distribution<double>(0., 1.)};
+    for (auto & manager : managers) {
+      selected_ids.emplace_back();
+      int ii{0};
+      for (auto center : manager) {
+        (void)center;  // to avoid compiler warning
+        double roll{dice(gen)};
```

Comment:
Is the rationale here that the _pseudo points_ are chosen randomly, because it is just a test? I am not sure I understand this test - the reference to which it is compared `sparse_points_b` are chosen how exactly from the representation?

Reply:
this tests picks randomly (for convinience) some pseudo points from a set of structures and serialize to json and deserialize the json and compare this new object with the original one.  I updated the test description, tell me if it makes more sense.